lib/core/test/unit/Cardano/Wallet/DBSpec.hs

```diff
@@ -747,7 +748,7 @@ dbPropertyTests = do
         it "Wallet Metadata"
             (checkCoverage . (prop_sequentialPut putWalletMeta readWalletMeta lrp))
         it "Tx History"
-            (checkCoverage . (prop_sequentialPut putTxHistoryF readTxHistoryF unions))
+            (checkCoverage . (prop_sequentialPut putTxHistoryF readTxHistoryF sortedUnions))
```

Comment:
So this was the one that failed previously because we were testing `getTxHistory(putTxHistory(x)) === x`, but needed to test `getTxHistory(putTxHistory(x)) === sorted x`?

Also, trying to understand… that what makes `prop_sequential_Put` sequential is just that we 1) share / don't clean the db, and 2) run multiple quickCheck tries?

Reply:
The property tests that after inserting several tx histories in the db, we read back a right-biased union of whatever has been inserted (in the sense that newly inserted stuff override the firstly inserted ones). Before, we returned a map so we didn't much care about ordering. Now, we have to check that the result is indeed sorted. 

It's sequential in opposition to parallel, where insertions would happen in any order. Not that there, we only compare the length of the result because we can't assume anything about the order of insertion. Still, we can expect the final number of inserted elements.

Reply:
Thanks!